src/Compilers/Core/Portable/Emit/EditAndContinue/DefinitionMap.cs

```diff
@@ -440,5 +473,139 @@ out IReadOnlyDictionary<(int syntaxOffset, AwaitDebugId debugId), StateMachineSt
+                            {
+                                yield return (IMethodSymbolInternal)synthesizedMember;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Enumerates method symbols synthesized for the body of a given <paramref name="oldMethod"/> in the previous generation that are not synthesized for the current method body (if any).
+        /// </summary>
+        /// <param name="oldMethod">Method from the previous generation.</param>
+        /// <param name="currentLambdas">Lambdas generated to the current version of the method. This includes both lambdas mapped to previous ones and newly introduced lambdas.</param>
+        public IEnumerable<IMethodSymbolInternal> GetDeletedSynthesizedMethods(IMethodSymbolInternal oldMethod, ImmutableArray<LambdaDebugInfo> currentLambdas)
+        {
+            var methodHandle = GetPreviousMethodHandle(oldMethod, out var peMethod);
+            var methodRowId = MetadataTokens.GetRowNumber(methodHandle);
+
+            if (Baseline.AddedOrChangedMethods.TryGetValue(methodRowId, out var addedOrChangedMethod))
+            {
+                // If a method has been added or updated then all synthesized members it produced are stored on the baseline.
+                // This includes all lambdas regardless of whether they were mapped to previous generation or not.
+                if (!addedOrChangedMethod.LambdaDebugInfo.IsDefaultOrEmpty &&
+                    Baseline.SynthesizedMembers.TryGetValue(oldMethod.ContainingType, out var synthesizedSiblingSymbols))
+                {
+                    return getSynthesizedClosureMethods(synthesizedSiblingSymbols, Baseline.SynthesizedMembers, addedOrChangedMethod.MethodId, addedOrChangedMethod.LambdaDebugInfo);
+                }
+
+                return SpecializedCollections.EmptyEnumerable<IMethodSymbolInternal>();
+            }
+
+            Debug.Assert(peMethod != null);
+
+            EditAndContinueMethodDebugInformation provider;
+            try
+            {
+                provider = Baseline.DebugInformationProvider(MetadataTokens.MethodDefinitionHandle(methodRowId));
+            }
+            catch (Exception e) when (e is InvalidDataException || e is IOException)
+            {
+                return SpecializedCollections.EmptyEnumerable<IMethodSymbolInternal>();
+            }
+
+            if (provider.Lambdas.IsDefaultOrEmpty)
+            {
+                return SpecializedCollections.EmptyEnumerable<IMethodSymbolInternal>();
+            }
+
+            var debugId = new DebugId(provider.MethodOrdinal, generation: 0);
+            return getSynthesizedClosureMethods(peMethod.ContainingType.GetMembers(), synthesizedMemberMap: null, debugId, provider.Lambdas);
+
+            IEnumerable<IMethodSymbolInternal> getSynthesizedClosureMethods(
```

Comment:
>getSynthesizedClosureMethods [](http://example.com/codeflow?start=47&length=28)

Should this be called "get**Deleted**SynthesizedClosureMethods" perhaps?

Reply:
The local function and the GetSynthesizedClosureMethods method are not specific to deleted methods. The find symbols that implement lambdas in general.

Reply:
Doesn't this local function return all the closure methods that are included in previous lambdas but not in current lambdas? (That sounds like lambdas that have been deleted, unless I'm misunderstanding.)

Reply:
You are right, I'll rename the local function.